File: csharp/src/Vista.SDK/LocationsDto.cs
using System.Text.Json.Serialization;

namespace Vista.SDK;

internal sealed record RelativeLocationsDto(
    [property: JsonPropertyName("code")] char Code,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("definition")] string? Definition
);

internal sealed record LocationsDto(
    [property: JsonPropertyName("visRelease")] string VisVersion,
    [property: JsonPropertyName("items")] RelativeLocationsDto[] Items
);

File: python/vista_sdk/LocationsDto.py
from typing import List, Optional

from pydantic import BaseModel, Field


class RelativeLocationsDto(BaseModel):
    code: str = Field(..., alias="code")
    name: str = Field(..., alias="name")
    definition: Optional[str] = Field(None, alias="definition")


class LocationsDto(BaseModel):
    vis_version: str = Field(..., alias="visRelease")
    items: List[RelativeLocationsDto] = Field(..., alias="items")
